utils/customize_node.py

```diff
@@ -48,16 +48,27 @@ def extract_bootfiles(args, keep_kernel=False):
     is an error (no way to choose programmatically).  Let the caller decide
     if zero kernels is an error.
 
-    :param 'args.build_dir': [str] where to move the unecessary files.
-    :param 'args.new_fs_dir': [str] where to find the unecessary files.
+    :param 'args' argparse.Namespace()
+        .build_dir: [str] where to move the unecessary files.
+        .new_fs_dir': [str] where to find the unecessary files.
+        .manifest.keep_kernel: user may choose for keep kernel in boot/
+
     :param 'keep_kernel': [bool] True - keep, but copy into the build dir.
                                 False - move kernel from boot/.
     :return: 'None' on success. Raise 'RuntimeError' on problems.
     """
     boot_dir = '%s/boot/' % args.new_fs_dir
     vmlinuz = glob.glob('%s/vmlinuz*' % (boot_dir))      # Move and keep
     initrd = glob.glob('%s/initrd.img*' % (boot_dir))    # Move and ignore
-    #also extract config-* and System.map* files installed with kernel.
+
+    is_force_keep_kernel = False
+    # 'manifest' param may not exist when building golden image.
+    if hasattr(args, 'manifest'):
+        #override 'keep_kernel' if user used 'keep_kernel' in the manifest
+        is_force_keep_kernel = getattr(args.manifest, 'keep_kernel', False)
+        keep_kernel = is_force_keep_kernel
```

Comment:
Why do you need two variables to track the same condition?  And where is "keep_kernel" set outside of this if statement?

Reply:
One is based of "args.hostname == golden" and the other one is coming from the manifest.
And now looking through it again, I don't like it.
The idea is: you should be able to control whether or not you want to keep kernel in boot or not.
For golden builds we always want to keep kernel in boot, but extract it during the customization stage.